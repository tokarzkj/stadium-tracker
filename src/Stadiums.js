import React from 'react';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import DeleteForever from '@material-ui/icons/DeleteForever';
const axios = require('axios');

class Stadiums extends React.Component {
    state = {
        stadiums: []
    }

    getStadiums() {
        axios.get('/stadium')
            .then((response) => {
                this.setState({stadiums: response.data});
            });
    }

    componentDidMount() {
        this.getStadiums();
    }

    render() {
        return (
            <Table>
                <TableHead>
                    <TableRow>
                        <TableCell>Stadium Name:</TableCell>
                        <TableCell>Visit Date:</TableCell>
                        <TableCell></TableCell>
                    </TableRow>
                </TableHead>
                <TableBody>
                    {this.state.stadiums.map(stadium => {
                        return (
                            <TableRow>
                                <TableCell>{stadium.stadiumName}</TableCell>
                                <TableCell>{new Date(stadium.visitDate).toLocaleDateString()}</TableCell>
                                <TableCell><DeleteForever/></TableCell>
                            </TableRow>
                        );
                    })}
                </TableBody>
            </Table>
        )
    }
}

export default Stadiums; 