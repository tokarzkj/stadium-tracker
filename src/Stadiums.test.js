import React from 'react';
import ReactDOM from 'react-dom';
import Stadiums from './Stadiums';
import { shallow } from 'enzyme';
import sinon from 'sinon';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Stadiums />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders correctly', () => {
    const fake = sinon.fake();
    sinon.replace(Stadiums.prototype, 'getStadiums', fake);
    const wrapper = shallow(<Stadiums />);
    wrapper.setState({ 
        stadiums: [{
            id: '1', 
            stadiumName: 'Ford Field', 
            visitDate: '2019-01-18 00:00:00'
        }]
    });
    
    expect(wrapper).toMatchSnapshot();
});