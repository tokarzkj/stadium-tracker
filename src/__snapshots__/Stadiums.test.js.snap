// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<WithStyles(Table)>
  <WithStyles(TableHead)>
    <WithStyles(TableRow)>
      <WithStyles(TableCell)>
        Stadium Name:
      </WithStyles(TableCell)>
      <WithStyles(TableCell)>
        Visit Date:
      </WithStyles(TableCell)>
      <WithStyles(TableCell) />
    </WithStyles(TableRow)>
  </WithStyles(TableHead)>
  <WithStyles(TableBody)>
    <WithStyles(TableRow)>
      <WithStyles(TableCell)>
        Ford Field
      </WithStyles(TableCell)>
      <WithStyles(TableCell)>
        1/18/2019
      </WithStyles(TableCell)>
      <WithStyles(TableCell)>
        <pure(DeleteForeverIcon)
          onClick={[Function]}
        />
      </WithStyles(TableCell)>
    </WithStyles(TableRow)>
  </WithStyles(TableBody)>
</WithStyles(Table)>
`;
